CAIPs/caip-222.md

```diff
@@ -0,0 +1,249 @@
+
+## Specification
+
+This JSON-RPC method can be requested to a wallet provider without prior knowledge of the blockchain accounts.
+
+The requester will provide parameters required by [CAIP-122][caip-122] plus a CACAO header type as specified by [CAIP-74][caip-74]
+
+The responder will return a signed CACAO with a header type and payload matching the requested parameters
+
+### Request
+
+The application would interface with a provider to make request as follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_authenticate",
+  "params": {
+      "cacaov": string,
+      "type": string,
+      "chainId": string,
+      "domain": string,
+      "aud": string,
+      "version": string,
+      "nonce": string,
+      "iat": string,
+      "exp": string, // optional
+      "nbf": string, // optional
+      "statement": string, // optional
+      "requestId": string, // optional
+      "resources": string[], // optional
+  }
+}
+```
+
+The JSON-RPC method is labelled as `wallet_authenticate` and expects the following parameters:
+
+- cacaov - Cacao version number
+- type - Cacao header message type
+- chainId - [CAIP-2][]-defined `chainId` to identify specific chain or network.
+- domain - [RFC 4501][rfc 4501] `dnsauthority` that is requesting the signing.
+- aud - [RFC 3986][rfc 3986] URI referring to the resource that is the subject of the signing.
+- version - Current version of the message.
+- nonce - Randomized token to prevent signature replay attacks.
+- iat - [RFC 3339][rfc 3339] `date-time` that indicates the issuance time.
+- exp (optional) - [RFC 3339][rfc 3339] `date-time` that indicates when the signed authentication message is no longer valid.
+- nbf (optional) - [RFC 3339][rfc 3339] `date-time` that indicates when the signed authentication message starts being valid.
+- statement (optional) - Human-readable ASCII assertion that the user will sign. It _MUST NOT_ contain `\n`.
+- requestId (optional) - System-specific identifier used to uniquely refer to the authentication request.
+- resources (optional) - List of information or references to information the user wishes to have resolved as part of the authentication by the relying party; express as [RFC 3986][rfc 3986] URIs and separated by `\n`.
+
+### Response
+
+The wallet will prompt the user with a dedicated UI to display the app requesting the authentication and allow the user to select which blockchain account to sign with.
+
+#### Success
+
+If approved, the walllet will return a signed CACAO with the selected blockchain account.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "h": {
+      "t": string,
+    },
+    "p": {
+      "iss": string,
+      "aud": string,
+      "exp": string,
+      "iat": string,
+      "nbf": string,
+      "nonce": string,
+      "domain": string,
+      "version": string,
+      "requestId": string,
+      "resources": string[],
```

Comment:
I prefer this being an strong array rather than a string, but we shouldn’t new line separate here if it’s going to be an array.

Reply:
I'm not sure I follow... isn't this the same interface for both SIWx and CACAO?

Reply:
It may be, I've been reviewing on mobile so wasn't too easy to double check SIWE/SIWX/CACAO specs. I think in SIWE the intention was to encode the entire message structure into a string which is where new line charters were needed. I'm not sure if that was kept in SIWX, but I saw above in the resources definition that we utilized the new line character as the list delimiter character. So, as it currently reads this could be interpreted as needing to have a new line character at the end of each string inside a string array which I don't think would be the intention. That was the main thing I wanted to call out.